### PR TITLE
Qlib data doc

### PR DESCRIPTION
**Pull request by chenditc**
Add doc and comments for data and feature 

## Description
Add comments to explain data structure and feature's meaning

## Motivation and Context
Documentation to explain data structure, what is the input and output for each step? Since most data in managed as file, user can sample a few to take a deeper look.

It would also be helpful for user to know how the price volume factor is calcualted and what do they mean.

## How Has This Been Tested?

- [ ] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [ ] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

## Types of changes

- [ ] Fix bugs
- [ ] Add new feature
- [ ] Update documentation

### CODE REVIEW

File: scripts/data_collector/yahoo/README.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ pip install -r requirements.txt
     - `target_dir`: save dir, by default *~/.qlib/qlib_data/cn_data*
     - `version`: dataset version, value from [`v1`, `v2`], by default `v1`
       - `v2` end date is *2021-06*, `v1` end date is *2020-09*
-      - user can append data to `v2`: [automatic update of daily frequency data](#automatic-update-of-daily-frequency-datafrom-yahoo-finance)
+      - If user want to incrementally update data, they need to use yahoo collector to [collect data from scratch](#collector-yahoofinance-data-to-qlib).
```
Comment (you-n-g) [inline scripts/data_collector/yahoo/README.md:39]:
```suggestion
      - If users want to incrementally update data, they need to use yahoo collector to [collect data from scratch](#collector-yahoofinance-data-to-qlib).
```

File: qlib/contrib/data/handler.py [hunk 2]
```diff
@@ -259,113 +259,163 @@ def parse_config_to_fields(config):
...
+                # The R-sqaure value of linear regression for the past d days, represent the trend linearity for past d days.
                 fields += ["Rsquare($close, %d)" % d for d in windows]
                 names += ["RSQR%d" % d for d in windows]
             if use("RESI"):
+                # The redisdual for linear regression for the past d days, represent the trend linearity for past d days. 
                 fields += ["Resi($close, %d)/$close" % d for d in windows]
                 names += ["RESI%d" % d for d in windows]
             if use("MAX"):
+                # The max price for past d days, divided by latest close price to remove unit
+                # Represent the upper resistent price range with 80% percentile of price.
```
Comment (you-n-g) [inline qlib/contrib/data/handler.py:292]: How can we get `with 80% percentile of price`  from the expression?

File: qlib/contrib/data/handler.py [hunk 3]
```diff
@@ -259,113 +259,163 @@ def parse_config_to_fields(config):
...
                 fields += ["Max($high, %d)/$close" % d for d in windows]
                 names += ["MAX%d" % d for d in windows]
             if use("LOW"):
+                # The low price for past d days, divided by latest close price to remove unit
+                # Represent the lower supporting price range with 20% percentile of price.
```
Comment (you-n-g) [inline qlib/contrib/data/handler.py:297]: How can we get `with 20% percentile of price` from the expression?

File: qlib/contrib/data/handler.py (hunk 2 above)
Comment (chenditc) [reply qlib/contrib/data/handler.py:292]: I was try to express this works with QTLU to for a resistence price range between 80% - 100% of price band. I guess this wording is too confusing. I will just remove this sentence.

**Comment by you-n-g** [comment]
Thanks so much. 
It looks great now.
The CI is broken due to the new version of PIP.
It will be fixed in https://github.com/microsoft/qlib/pull/1222/files


Welcome to become one of Qlib's [contributors](https://github.com/microsoft/qlib/graphs/contributors)

**Pull request merged**